### PR TITLE
Convert UniqueID::nil() to a constructor

### PR DESCRIPTION
**Pull request by stephanie-wang**
## What do these changes do?

The default `UniqueID()` does not initialize the ID data, which can make statements like `ObjectID object_id;` error-prone. This merges the `UniqueID::nil()` static method into the default constructor.

### CODE REVIEW

**Review by pcmoritz — APPROVED** [review]
LGTM conditionally on tests (esp. linting) passing.

**Review by robertnishihara — APPROVED** [review]
This looks good to me.

File: src/ray/raylet/object_manager_integration_test.cc [hunk 1]
```diff
@@ -206,7 +206,7 @@ class TestObjectManagerIntegration : public TestObjectManagerBase {
                   << "\n";
     ClientTableDataT data;
     gcs_client_2->client_table().GetClient(client_id_1, data);
-    RAY_LOG(INFO) << (ClientID::from_binary(data.client_id) == ClientID::nil());
+    RAY_LOG(INFO) << (ClientID::from_binary(data.client_id) == ClientID());
```
Comment (pcmoritz) [inline src/ray/raylet/object_manager_integration_test.cc]: above we used .is_nil() for this, we might want to unify for consistency

Comment (stephanie-wang) [reply src/ray/raylet/object_manager_integration_test.cc]: Thanks!

**Comment by raulchen** [comment]
Actually I prefer keeping the `UniqueID::nil()` method, for 2 reasons:
1) compared to `UniqueID()`, the meaning of `UniqueID::nil()` more clear. Users know this method returns a nil ID, without needing to refer to the document.
2) With factory method `UniqueID::nil()`, we can actually cache one NIL instance for all use cases (assuming `UniqueID` is immutable). Compared to creating a new instance every time, this is more efficient. 

But, it's good to initialize data to nil in the constructor.

**Pull request merged**